pkg/module/manager.go

```diff
@@ -81,3 +99,93 @@ func (m *ManagerImpl) ExampleConfigs() []string {
 	}
 	return ret
 }
+
+var errNegotiationFailed = fmt.Errorf("version mismatch between module and providers, please upgrade your provider and/or cloudquery")
+
+func (m *ManagerImpl) negotiateProtocolVersions(ctx context.Context, mod Module, provs []*cqproto.GetProviderSchemaResponse, forceVersion uint32) (uint32, map[string]ProviderData, error) {
+	var doVersions []uint32
+	if forceVersion > 0 {
+		doVersions = []uint32{forceVersion}
+	} else {
+		doVersions = mod.ProtocolVersions()
+	}
+
+	var (
+		ret               = make(map[string]ProviderData, len(provs)) // provider vs. info-key vs. files provided by provider under that key
+		supportedVersions = make(map[string][]uint32, len(provs))
+		foundUnsupported  bool
+	)
+
+	// Do initial requests
+	for _, p := range provs {
```

Comment:
I think the negotiation should be in the rpc level and not need for two loops, i.e give me versions X, and you get them back. Moreover we can do that request in the client and pass the data in the execRequest (the module info). If we don't support any versions prior to the call we shouldn't call the execute. (It couples the client with the modules a little, but the issue here is the general design of the client, and I don't think we should go deep and refactor that now).